src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,90 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Calculator {
+
+    private final String customDelimiterPattern = "//(.)\n(.*)";
+    private final String defaultDelimiterPattern = ",|:";
+
+    public int splitAndSum(String input) {
+        if (isInputNullOrEmpty(input)) {
+            return 0;
+        }
+        if (CustomSeparatorExists(input)) {
+            return calculateWithCustomDelimiter(input);
+        }
+        if (isInputOneNumberString(input)) {
+            return calcuateOneNumberInputString(input);
+        }
+        return calculateWithDefaultSeparator(input);
+    }
```

Comment:
조건문 조건에서 호출하는 메서드들이 같은 클래스 내에 중구난방으로 펴져있는대 이러면 해당 메서드들을 파악하기 위해 클래스의 여러 부분들을 왔다 갔다 해야 되서 혼란을 야기해요.
[클린코드 챕터 5. 형식 맞추기](https://www.skullkim-dev.com/2022/01/13/5.-%ED%98%95%EC%8B%9D-%EB%A7%9E%EC%B6%94%EA%B8%B0/) 에서 '세로 밀집도' ~ '세로 순서' 부분을 모두 읽어보세요~

단지 테스트를 위해 private으로 두어야 하는 메서드를 public으로 두는 것은 좋지 않아요. private으로 두어야 하는 메서드는 어떤 의미를 가질까요? 유닛이 맞을까요?

```suggestion
    public int splitAndSum(String input) {
        if (isInputNullOrEmpty(input)) {
            return 0;
        }
        if (CustomSeparatorExists(input)) {
            return calculateWithCustomDelimiter(input);
        }
        if (isInputOneNumberString(input)) {
            return calcuateOneNumberInputString(input);
        }
        return calculateWithDefaultSeparator(input);
    }
```

Reply:
'세로 밀집도' ~ '세로 순서' 부분을 읽고 나서 생각한 개선방법은 호출되는 순서대로 함수들을 배치하는 것인데 윤기님이 의도하신 부분이 맞을까요?
이렇게 되면 클래스의 여러부분을 왔다갔다 하지 않아도 될 것 같아서 가독성이 많이 좋아질 것 같아요.

Reply:
테스트를 위해 private으로 두어야 하는 메서드를 public으로 두는 것은 좋지 않다고 하신 부분을 개선하려고 스스로 생각해봤는데 많이 어렵게 느껴져지더라구요. 구글링으로 관련된 좋은 글을 찾아 읽어보고 개선방안을 알게 되었어요.
링크: https://mangkyu.tistory.com/235

private 메소드를 테스트하기 위해 public으로 두는 방식은 잘못된 것 같아요.
public으로 두지 않고 reflection을 이용해 테스트하는 방식이 있더라구요.
하지만 reflection을 이용해 private 메소드를 테스트하는 것을 지양해야 할 것 같아요.

private으로 두어야 하는 메서드는 유닛에 포함하지 않고
private 메소드를 호출하는 public 메소드로 private 메소드 로직을 테스트해야 할 것 같은데 이 방식이 맞을까요?

Reply:
맞아요😄